### DISCUSSION TITLE
Docker app and cron separated

### DISCUSSION THREAD

**Discussion by enboig** [firefly-iii/firefly-iii#9358] [General]
Is there a reason to run cron com a separate service (and using wget)?

Wouldn't be more simple (and avoid possible timeouts) running it in `app` service using `php artisan`?

I know not all hostings may allow running cron and cli commands, but it feels odd to have a whole service running just to execute a job daily.

**Comment by JC5** [comment]
Hey, great question. 

Separation of concerns is important in docker containers. You can only run one process in a container. 

I used to use "supervisord" to run multiple things in one container, but that's a lot of maintenance. 

This is cleaner and gives the user more freedom. 

Hope this helps :)

**Comment by enboig** [comment]
What about a `cron` image based in `app` image, where "wget..." becomes "php artisan ..."? it could access the same variables/database/files/mountpoints/... than app

**Reply by JC5** [reply]
No. But you don't have to use this cron container to do cron.

**Reply by enboig** [reply]
I know it is a simple helper container to load cons pages; I was just looking for a _better_ way of doing it. I remember developing long ago with drupal that some sites run into time outs or memory limit errors when running cron through apache, and most advices where to run cron using `drush` (a console command).

I managed to enter the running container `docker compose exec -it app sh` and run
```
# php artisan schedule:run                                                                                                                                                                          
                                                                                                                                                                                                    
   INFO  No scheduled commands are ready to run.                                                                                                                                                    
```

But it is not clear to me if this is the same as `wget ...`

**Reply by JC5** [reply]
https://docs.firefly-iii.org/how-to/firefly-iii/advanced/cron/

Please make sure you run the correct command.

**Comment by enboig** [comment]
I will check de documentation. Thanks for your time.
